woocommerce/woocommerce
# Add initial migration script

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR is the first of several for the migration/integration script to migrate plugins to the Blocks paradigm.
This initial PR creates a new package, and adds the initial script. For now it only outputs the plugin information if a plugin file exists.

Closes #40311  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Load this branch and run `pnpm run build`
2. Go to the `packages/js/integrate-plugin` folder
3. Run `npm link`
4. Now either create a test plugin in another folder or download another plugin like [Hello Dolly ](https://wordpress.org/plugins/hello-dolly/)
5. Go to the plugin folder and run `npx woo-integrate-plugin`
6. It should console log a JSON file with the plugin data: `name`, `description`, `version`, `textdomain` and `namespace` it also logs out an empty object for the plugin config ( which is irrelevant for now ).
7. Try editing the main plugin file a bit, adding or removing some spaces in the plugin header description comment.
8. Run `npx woo-integrate-plugin` again, it should output the plugin data correctly with the data trimmed.
9. Now run it in a empty folder or a folder that is not a plugin
10. It should output `Plugin file not found.`



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
